src/server.cpp

```diff
@@ -711,6 +720,26 @@ Response InternalServer::handle_random(const RequestContext& request)
   }
 }
 
+Response InternalServer::handle_captured_external(const RequestContext& request)
+{
+  std::string source = "";
+  try {
+    source = kiwix::urlDecode(request.get_argument("source"));
+  } catch (const std::out_of_range& e) {}
+
+  if (source.empty())
+    return build_404(request, "");
+
+  auto data = get_default_data();
+  data.set("source", source);
+  auto response = get_default_response();
```

Comment:
Instead of `get_default_response`, we could simply directly create the `Response` with `blockExternalLinks`==false.

This way, we would not have to change the signature of `set_taskbar`

Reply:
Did that then came back to default without taskbar call as not required anymore (page detected in JS)